applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -27,30 +27,39 @@ def __init__(self,model,parameters):
         else:
             self.is_printing_rank = True
 
-        ## Create model part and solver (but don't initialize them yet)
-        model_part_name = self.project_parameters["problem_data"]["model_part_name"].GetString()
-        self.main_model_part = Kratos.ModelPart(model_part_name)
+        # Deprecation warnings
+        solver_settings = self.project_parameters["solver_settings"]
+        if not solver_settings.Has("domain_size"):
+            Kratos.Logger.PrintInfo("FluidDynamicsAnalysis", "Using the old way to pass the domain_size, this will be removed!")
+            solver_settings.AddEmptyValue("domain_size")
+            solver_settings["domain_size"].SetInt(self.project_parameters["problem_data"]["domain_size"].GetInt())
+
+        if not solver_settings.Has("model_part_name"):
+            Kratos.Logger.PrintInfo("FluidDynamicsAnalysis", "Using the old way to pass the model_part_name, this will be removed!")
+            solver_settings.AddEmptyValue("model_part_name")
+            solver_settings["model_part_name"].SetString(self.project_parameters["problem_data"]["model_part_name"].GetString())
 
         import python_solvers_wrapper_fluid
-        self.solver = python_solvers_wrapper_fluid.CreateSolver(self.main_model_part, self.project_parameters)
+        self.solver = python_solvers_wrapper_fluid.CreateSolver(model, self.project_parameters)
+
+        self.__restart_utility = None
 
     def Initialize(self):
         '''
         Construct and initialize all classes and tools used in the simulation loop.
         '''
-        domain_size = self.project_parameters["problem_data"]["domain_size"].GetInt()
-        self.main_model_part.ProcessInfo.SetValue(Kratos.DOMAIN_SIZE, domain_size)
 
         self._SetUpRestart()
 
         if self.load_restart:
-            self.restart_utility.LoadRestart()
+            restart_utility = self._GetRestartUtility()
+            restart_utility.LoadRestart()
```

Comment:
this is in the baseclass now

Also I am writing a process for saving restarts atm such that one does not have to manually include it in each stage

Reply:
I think this will come in #2135? I'll update once that is merged, then.